src/linearAlgebra/BlasLapackKernels.cpp

```diff
@@ -79,12 +95,14 @@ namespace dftefe
 
 
 
-      template <typename ValueType, dftefe::utils::MemorySpace memorySpace>
+      template <dftefe::utils::MemorySpace memorySpace,
+                typename ValueType1,
+                typename ValueType2>
       std::vector<double>
-      Kernels<ValueType, memorySpace>::nrms2MultiVector(
+      Kernels<memorySpace, ValueType1, ValueType2>::nrms2MultiVector(
```

Comment:
Same as the above comment. It should just be `template <utils::MemorySpace memorySpace, typename ValueType1> Kernels<memorySpace, ValueType1>::nrmsMulitVector()`